pkg/pushsync/pushsync.go

```diff
@@ -563,41 +541,36 @@ func (ps *PushSync) pushPeer(ctx context.Context, skip *skippeers.List, resultCh
 		return
 	}
 
-	err = creditAction.Apply()
+	err = action.Apply()
 }
 
-func (ps *PushSync) pushToNeighbourhood(ctx context.Context, skiplist []swarm.Address, ch swarm.Chunk, origin bool, originAddr swarm.Address) {
+// pushToNeighbourhood pushes the chunk to some peers in the neighborhood in parallel for replication.
+func (ps *PushSync) pushToNeighbourhood(ctx context.Context, skiplist []swarm.Address, ch swarm.Chunk, origin bool) {
 	count := 0
-	// Push the chunk to some peers in the neighborhood in parallel for replication.
-	// Any errors here should NOT impact the rest of the handler.
 	_ = ps.topologyDriver.EachConnectedPeer(func(peer swarm.Address, po uint8) (bool, bool, error) {
-		// skip forwarding peer
-		if peer.Equal(originAddr) {
-			return false, false, nil
-		}
 
 		// skip skiplisted peers
 		if swarm.ContainsAddress(skiplist, peer) {
 			return false, false, nil
 		}
 
-		// here we skip the peer if the peer is closer to the chunk than us
-		// we replicate with peers that are further away than us because we are the storer
-		if closer, _ := peer.Closer(ch.Address(), ps.address); closer {
-			return false, false, nil
+		// stop here since the rest of the peers will also be further away
+		if po < ps.radiusChecker.StorageRadius() {
+			return true, false, nil
 		}
 
 		if count == nPeersToReplicate {
 			return true, false, nil
 		}
+
 		count++
-		go ps.pushToNeighbour(ctx, peer, ch, origin)
+		go ps.replicateWithPeer(ctx, peer, ch, origin)
 		return false, false, nil
 	}, topology.Filter{Reachable: true})
 }
 
-// pushToNeighbour handles in-neighborhood replication for a single peer.
-func (ps *PushSync) pushToNeighbour(ctx context.Context, peer swarm.Address, ch swarm.Chunk, origin bool) {
+// replicateWithPeer handles in-neighborhood replication for a single peer.
+func (ps *PushSync) replicateWithPeer(ctx context.Context, peer swarm.Address, ch swarm.Chunk, origin bool) {
```

Comment:
`replicateWithPeer` seems to be doing very similar to `pushChunk`, would it be possible to avoid duplication?